cypress/integration/rich-text/RichTextEditor.spec.ts

```diff
@@ -658,6 +658,47 @@ describe('Rich Text Editor', { viewportHeight: 2000 }, () => {
       ].map((type) => getDropdownItem(type).get('button').should('not.be.disabled'));
     });
 
+    describe('Inserting Tables', () => {
+      it('replaces empty paragraphs with tables', () => {
+        insertTable();
+
+        richText.expectValue(
+          doc(
+            table(row(emptyHeader(), emptyHeader()), row(emptyCell(), emptyCell())),
+            emptyParagraph()
+          )
+        );
+      });
+
+      it('inserts new table below if paragraph is not empty', () => {
+        richText.editor.type('foo');
+
+        insertTable();
+
+        richText.expectValue(
+          doc(
+            paragraphWithText('foo'),
+            table(row(emptyHeader(), emptyHeader()), row(emptyCell(), emptyCell())),
+            emptyParagraph()
+          )
+        );
+      });
+
+      describe('Toolbar', () => {
+        const buttonsToDisableTable = ['quote', 'ul', 'ol'];
+
+        buttonsToDisableTable.forEach((button) => {
```

Comment:
I think we can be more pragmatic and test all of these buttons in one test to have a positive impact on their runtime.

```ts
it('disables toolbar buttons for elements unsupported within tables')
```

There's also `hr` which is not allowed within a table and you can also assert that the embeds button turns into "Embed inline entry" only.

Reply:
We have tests for the second case you've mentioned: https://github.com/contentful/field-editors/blob/next/cypress/integration/rich-text/RichTextEditor.spec.ts#L622

I really don't have a strong opinion against having that in one test or not.